dask_sql/physical/rel/logical/aggregate.py

```diff
@@ -229,18 +229,15 @@ def convert(self, rel: "LogicalPlan", context: "dask_sql.Context") -> DataContai
             logger.debug("Performing full-table aggregation")
 
         # Do all aggregates
-        df_result, output_column_order, cc = self._do_aggregations(
+        df_agg, output_column_order, cc = self._do_aggregations(
             rel,
             dc,
             group_columns,
             context,
         )
 
-        # SQL does not care about the index, but we do not want to have any multiindices
-        df_agg = df_result.reset_index(drop=True)
-
-        # Fix the column names and the order of them, as this was messed with during the aggregations
-        df_agg.columns = df_agg.columns.get_level_values(-1)
+        # SQL does not care about the index, but if group columns were specified we'll want to keep those
+        df_agg = df_agg.reset_index(drop=(not group_columns))
```

Comment:
In cases where we have multiple groupby columns but are selecting only 1 does it make sense to handle that logic of dropping the rest here or do we rely on the `limit_to` projection mechanisms downstream to achieve this?

Reply:
Yeah right now we're relying on `limit_to` to exclude the unnecessary columns, not sure if it's worth it to manually drop these columns after the `reset_index`